bindings/c/test/mako/mako.cpp

```diff
@@ -1415,132 +1444,28 @@ void printStatsHeader(Arguments const& args, bool show_commit, bool is_first_hea
-	const auto tps_i = static_cast<uint64_t>(tps_f);
-	fmt::printf("Total Xacts:       %8lu\n", final_stats.getOpCount(OP_TRANSACTION));
-	fmt::printf("Total Conflicts:   %8lu\n", final_stats.getConflictCount());
-	fmt::printf("Total Errors:      %8lu\n", final_stats.getTotalErrorCount());
-	fmt::printf("Overall TPS:       %8lu\n\n", tps_i);
-
-	if (fp) {
-		fmt::fprintf(fp, "\"results\": {");
-		fmt::fprintf(fp, "\"totalDuration\": %6.3f,", duration_sec);
-		fmt::fprintf(fp, "\"totalProcesses\": %d,", args.num_processes);
-		fmt::fprintf(fp, "\"totalThreads\": %d,", args.num_threads);
-		fmt::fprintf(fp, "\"totalAsyncXacts\": %d,", args.async_xacts);
-		fmt::fprintf(fp, "\"targetTPS\": %d,", args.tpsmax);
-		fmt::fprintf(fp, "\"totalXacts\": %lu,", final_stats.getOpCount(OP_TRANSACTION));
-		fmt::fprintf(fp, "\"totalConflicts\": %lu,", final_stats.getConflictCount());
-		fmt::fprintf(fp, "\"totalErrors\": %lu,", final_stats.getTotalErrorCount());
-		fmt::fprintf(fp, "\"overallTPS\": %lu,", tps_i);
-	}
-
-	/* per-op stats */
-	printStatsHeader(args, true, true, false);
-
-	/* OPS */
-	putTitle("Total OPS");
-	if (fp) {
-		fmt::fprintf(fp, "\"totalOps\": {");
-	}
-	auto first_op = true;
-	for (auto op = 0; op < MAX_OP; op++) {
-		if ((args.txnspec.ops[op][OP_COUNT] > 0 && op != OP_TRANSACTION) || op == OP_COMMIT) {
-			putField(final_stats.getOpCount(op));
-			if (fp) {
-				if (first_op) {
-					first_op = 0;
-				} else {
-					fmt::fprintf(fp, ",");
-				}
-				fmt::fprintf(fp, "\"%s\": %lu", getOpName(op), final_stats.getOpCount(op));
-			}
-		}
-	}
-
-	/* TPS */
-	const auto tps = final_stats.getOpCount(OP_TRANSACTION) / duration_sec;
-	putFieldFloat(tps, 2);
-
-	/* Conflicts */
-	const auto conflicts_rate = final_stats.getConflictCount() / duration_sec;
-	putFieldFloat(conflicts_rate, 2);
-	fmt::print("\n");
-
-	if (fp) {
-		fmt::fprintf(fp, "}, \"tps\": %.2f, \"conflictsPerSec\": %.2f, \"errors\": {", tps, conflicts_rate);
-	}
+void printThreadStats(ThreadStatistics& final_stats, Arguments& args, FILE* fp, bool is_report = false) {
 
-	/* Errors */
-	putTitle("Errors");
-	first_op = 1;
-	for (auto op = 0; op < MAX_OP; op++) {
-		if (args.txnspec.ops[op][OP_COUNT] > 0 && op != OP_TRANSACTION) {
-			putField(final_stats.getErrorCount(op));
-			if (fp) {
-				if (first_op) {
-					first_op = 0;
-				} else {
-					fmt::fprintf(fp, ",");
-				}
-				fmt::fprintf(fp, "\"%s\": %lu", getOpName(op), final_stats.getErrorCount(op));
+	if (is_report) {
+		for (auto op = 0; op < MAX_OP; op++) {
+			if (final_stats.getLatencySampleCount(op) > 0 && op != OP_COMMIT && op != OP_TRANSACTION) {
+				args.txnspec.ops[op][OP_COUNT] = 1;
 			}
 		}
 	}
-	if (fp) {
-		fmt::fprintf(fp, "}, \"numSamples\": {");
-	}
-	fmt::print("\n\n");
 
 	fmt::print("Latency (us)");
 	printStatsHeader(args, true, false, true);
 
 	/* Total Samples */
 	putTitle("Samples");
-	first_op = 1;
+	auto first_op = 1;
```

Comment:
Should declare this type as an int. I prefer auto to be used when getting a complex type from a function.

Reply:
So this code is copied from before. I just moved it into a new function, but I'll go ahead and change it to bool and use true/false.